homeassistant/components/modbus/binary_sensor.py

```diff
@@ -109,9 +111,16 @@ async def async_update(self, now: datetime | None = None) -> None:
             self._result = None
         else:
             self._lazy_errors = self._lazy_error_count
-            self._attr_is_on = result.bits[0] & 1
             self._attr_available = True
             self._result = result
+            if self._input_type in (CALL_TYPE_COIL, CALL_TYPE_DISCRETE):
+                self._attr_is_on = bool(result.bits[0] & 1)
+            else:
+                value = int(result.registers[0])
```

Comment:
The holding register only holds 1 bit ??? I would have expected it to contain 16bits just like the coil and discrete_input.

Reply:
Updated to behave the same way as Coil/Discrete